tools/2025-01-gitops-technical-design.md

```diff
@@ -0,0 +1,435 @@
+
+## Design
+
+### High Level Design
+<!--
+High level overview of the data flow and key components.
+
+Provide a high-level description, using diagrams as appropriate, and top-level
+explanations to convey the architectural/design overview. Don’t go into a lot
+of details yet but provide enough information about the relationship between
+these components and other components. Call out or highlight new components
+that are not part of this feature (dependencies). This diagram generally
+treats the components as black boxes. Provide a pointer to a more detailed
+design document, if one exists.
+-->
+
+The Radius Flux Controller reads the Git repository, compiles Bicep files, and creates DeploymentTemplate resources on the cluster. It interacts with the DeploymentTemplate controller to ensure that the desired state defined in the Git repository is accurately reflected in the Kubernetes cluster.
+
+### Architecture Diagram
+<!--
+Provide a diagram of the system architecture, illustrating how different
+components interact with each other in the context of this proposal.
+
+Include separate high level architecture diagram and component specific diagrams, wherever appropriate.
+-->
+
+![Architecture Diagram Part 1](2025-01-gitops-technical-design/rfc-1.png)
+
+![Architecture Diagram Part 2](2025-01-gitops-technical-design/rfc-2.png)
+
+### Detailed Design: Radius Flux Controller
+
+<!--
+This section should be detailed and thorough enough that another developer
+could implement your design and provide enough detail to get a high confidence
+estimate of the cost to implement the feature but isn’t as detailed as the
+code. Be sure to also consider testability in your design.
+
+For each change, give each "change" in the proposal its own section and
+describe it in enough detail that someone else could implement it. Cover
+ALL of the important decisions like names. Your goal is to get an agreement
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+Discuss the rationale behind architectural choices and alternative options
+considered during the design process.
+-->
+
+The Radius Flux Controller will be implemented as a new controller in the `controller` service. Here is the high-level control loop of the new controller:
+
+1. The controller lists all `GitRepository` resources in the cluster.
+1. For each `GitRepository`, the controller parses the contents of the Git repository.
+1. The controller checks for a `radius-config.yaml` file in the repository. If the file is not present, then the controller skips the repository.
```

Comment:
Will this be logged to the user that the repo was skipped?

Reply:
It will be logged in the radius controller logs